### PR TITLE
AST-3305 - Fix:Display ruleset not working as expected on singular post/tag/page

### PR DESCRIPTION
**Pull request by yssambare12**
**Description** - Display rules not working as excepted in the sidebar manager plugin. 

**Steps To Reproduce:** 

1. I created a Category named "AAA" and assigned the post titled "Hello World" to the "AAA" category.
2. Additionally, I assigned a test sidebar to be displayed for all singular posts from the "AAA" category.
3. As a result, the test sidebar should be visible on the URL  .
4. However, when I revisit the editing page for the "Test Sidebar" and simply click "Update," the following issues occur:
5. The test sidebar appears differently than intended.
6. On the front end of the website, the "Test Sidebar" disappears completely.
7. It is worth noting that the "Display On" setting is being saved as "tax-2" instead of "tax-2-single-category."

**Threads**

1. https://wordpress.org/support/topic/all-singulars-from-category-turn-into-category-category/page/2/
2. https://wordpress.org/support/topic/all-singulars-from-category-turn-into-category-category-2/

### CODE REVIEW

File: classes/modules/target-rule/class-bsf-sb-target-rules-fields.php [hunk 1]
```diff
@@ -681,11 +681,29 @@ public static function generate_target_rule_selector( $type, $selection_options,
...
+						
+							$tax_id = (int) str_replace( 'tax-', '', $sel_value );
+						
+							$term = get_term( $tax_id );
+						
 							$term_taxonomy = ucfirst( str_replace( '_', ' ', $term->taxonomy ) );
-							$output       .= '<option value="tax-' . $tax_id . '" selected="selected" >' . $term->name . ' - ' . $term_taxonomy . '</option>';
-
+						
+							$output .= '<option value="tax-' . $tax_id . $suffix . '" selected="selected" >' . $term->name . ' - ' . $term_taxonomy . $term_suffix . '</option>';
```
Comment (imnavanath) [inline classes/modules/target-rule/class-bsf-sb-target-rules-fields.php:703]: As we are changing value the selected will work in the case on users end with their old saved value??

Comment (yssambare12) [reply classes/modules/target-rule/class-bsf-sb-target-rules-fields.php:703]: @imnavanath No! modification which I did here will not update or affected the selected values for existing user save values; they need to update their save values to reflect the changes.

Comment (imnavanath) [reply classes/modules/target-rule/class-bsf-sb-target-rules-fields.php:703]: If suppose, I have selected a value in master version and when I'll switch to this update the select will come with my last selected value only or deselect my previous value? Also check & resolve codesniffer failed GitHub pipeline.

Comment (yssambare12) [reply classes/modules/target-rule/class-bsf-sb-target-rules-fields.php:703]: Yes @imnavanath change come with latest selected value for this perticluar Singulars case, & rest selecetd sidebar/values will remain same .

Comment (yssambare12) [reply classes/modules/target-rule/class-bsf-sb-target-rules-fields.php]: Resolved odesniffer failed GitHub pipeline.

**Review by imnavanath — APPROVED** [review]

**Pull request merged**